File: 0826-most-profit-assigning-work/0826-most-profit-assigning-work.java
class Solution {
    public int maxProfitAssignment(int[] difficulty, int[] profit, int[] worker) {
        int n = profit.length;
        List<Pair<Integer, Integer>> list = new ArrayList<>();
        for(int i=0; i<n; i++) {
            list.add(new Pair<>(difficulty[i], profit[i]));
        }
        Collections.sort(list, Comparator.comparingInt(Pair::getKey));
        Arrays.sort(worker);
        int ind = 0, mx = 0, ans = 0;
        for(int i=0; i<worker.length; i++) {
            while(ind < n && worker[i] >= list.get(ind).getKey()) {
                mx = Math.max(mx, list.get(ind).getValue());
                ind++;
            }
            ans += mx;
        }
        return ans;
    }
}

File: 0826-most-profit-assigning-work/0826-most-profit-assigning-work.cpp
class Solution {
public:
    int maxProfitAssignment(vector<int>& difficulty, vector<int>& profit, vector<int>& worker) {
        int ans = 0, n = profit.size();
        vector<pair<int,int>>vp;
        for(int i=0; i<n; i++) {
            vp.push_back({difficulty[i], profit[i]});
        }
        sort(vp.begin(), vp.end());
        sort(worker.begin(), worker.end());
        int ind = 0, mx = 0;
        for(int i=0; i<worker.size(); i++) {
            while(ind < n && worker[i] >= vp[ind].first) {
                mx = max(mx, vp[ind].second);
                ind++;
            }
            ans += mx;
        }
        return ans;
    }
};